### PR TITLE
Mantic merge 2024 02 13

### PR DESCRIPTION
**Pull request by Chris-Peterson444**
- cherry-picked 9ecc406 from #1911
 - Related: LP:#2052524

### CODE REVIEW

**Comment by Chris-Peterson444** [comment]
Updated the context manager syntax to by python3.8 compliant

File: subiquity/server/controllers/tests/test_network.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        Since netplan 0.106.1, Netplan YAMLs should have file
+        permissions with mode 0o600 (root/owner RW only).
+        """
+
+        with patch("os.getuid", return_value=0):
+            with patch("os.chmod") as mock_chmod:
+                with patch("os.chown") as mock_chown:
+                    self.controller._write_config()
+                    mock_chmod.assert_called_with(ANY, 0o600)
```
Comment (ogayot) [inline subiquity/server/controllers/tests/test_network.py:47]: completely minor (don't change it now!), but you can move the `assert` statements outside of the with blocks :)

**Review by ogayot — APPROVED** [review]
LGTM

**Comment by murraybd** [comment]
As a member of the Ubuntu Release team I approve these changes.

**Pull request merged**